src/test/java/co/rsk/federate/FederationProviderFromFederatorSupportTest.java

```diff
@@ -370,9 +371,159 @@ void getRetiringFederation_present_p2sh_erp_federation() {
         assertEquals(expectedFederationAddress, obtainedFederation.getAddress());
     }
 
+    @Test
+    void getProposedFederation_whenRSKIP417IsNotActivated_shouldReturnEmptyOptional() {
+        // Arrange
+        ActivationConfig.ForBlock configMock = mock(ActivationConfig.ForBlock.class);
+        when(configMock.isActive(RSKIP419)).thenReturn(false);
+        when(federatorSupportMock.getConfigForBestBlock()).thenReturn(configMock);
+
+        // Act & Assert
+        assertEquals(Optional.empty(), federationProvider.getProposedFederation());
+    }
+
+    @Test
+    void getProposedFederation_whenProposedFederationSizeIsNonExistent_shouldReturnEmptyOptional() {
+        // Arrange
+        ActivationConfig.ForBlock configMock = mock(ActivationConfig.ForBlock.class);
+        when(configMock.isActive(RSKIP419)).thenReturn(true);
+        when(federatorSupportMock.getConfigForBestBlock()).thenReturn(configMock);
+        when(federatorSupportMock.getProposedFederationSize())
+            .thenReturn(Optional.of(FEDERATION_NON_EXISTENT.getCode()));
+
+        // Act & Assert
+        assertEquals(Optional.empty(), federationProvider.getProposedFederation());
+        verify(federatorSupportMock).getProposedFederationSize();
+        verify(federatorSupportMock).getProposedFederationAddress();
+    }
+
+    @Test
+    void getProposedFederation_whenProposedFederationAddressDoesNotExist_shouldReturnEmptyOptional() {
+        // Arrange
+        ActivationConfig.ForBlock configMock = mock(ActivationConfig.ForBlock.class);
+        when(configMock.isActive(RSKIP419)).thenReturn(true);
+        when(federatorSupportMock.getConfigForBestBlock()).thenReturn(configMock);
+        when(federatorSupportMock.getProposedFederationSize())
+            .thenReturn(Optional.of(9));
+
+        // Act & Assert
+        assertEquals(Optional.empty(), federationProvider.getProposedFederation());
+        verify(federatorSupportMock).getProposedFederationSize();
+        verify(federatorSupportMock).getProposedFederationAddress();
+    }
+
+    @Test
+    void getProposedFederation_whenExistsAndIsStandardMultisigFederation_shouldReturnProposedFederation() {
```

Comment:
As said above, once `RSKIP419`, all pending and proposed federations are always `P2shFeds`. Therefore, this is an unrealistic scenario.

Reply:
Removed, as well as the other one